src/components/SkillCreator/SkillCreator.js

```diff
@@ -33,95 +33,383 @@ import ChevronRight from 'material-ui/svg-icons/navigation/chevron-right';
       anchorOrigin: {
         horizontal: 'left',
         vertical: 'bottom',
       },
+      groups: [],
+      loadViews: false,
+      code: skillBuildCode,
+      editable: !!cookies.get('loggedIn'),
     };
+
+    if (this.props.location.pathname.split('/')[1] === 'skillCreator') {
+      this.state = {
+        ...commonState,
+        mode: 'create',
+        codeView: true,
+        conversationView: false,
+        treeView: false,
+        showImage: false,
+        loading: false,
+        imageUrl: '<image_url>',
+        commitMessage: '',
+        modelValue: null,
+        groupValue: null,
+        languageValue: null,
+        groupSelect: true,
+        languageSelect: true,
+        expertSelect: true,
+        expertValue: '',
+        slideState: 1, // 1 means in middle, 2 means preview collapsed
+        colSkill: 8,
+        colPreview: 4,
+        prevButton: 0, // 0 means disappear, 1 means appear
+      };
+    } else {
+      if (this.props.hasOwnProperty('revertingCommit')) {
+        this.setState({
+          commitMessage: 'Reverting to commit - ' + this.props.revertingCommit,
+        });
+      } else if (this.props.location.pathname.split('/')[5]) {
+        this.setState({
+          commitMessage:
+            'Reverting to commit - ' +
+            this.props.location.pathname.split('/')[5],
+        });
+      }
+      this.state = {
+        ...commonState,
+        mode: 'edit',
+        codeView: true,
+        conversationView: false,
+        treeView: false,
+        showImage: true,
+        loading: false,
+        image: '',
+        skillUrl: null,
+        commitMessage: `Updated Skill ${
+          this.props.location.pathname.split('/')[2]
+        }`,
+        modelValue: 'general',
+        codeChanged: false,
+        showDeleteBox: false,
+        groupValue: this.props.location.pathname.split('/')[1],
+        oldGroupValue: this.props.location.pathname.split('/')[1],
+        languageValue: this.props.location.pathname.split('/')[4],
+        oldLanguageValue: this.props.location.pathname.split('/')[4],
+        expertValue: this.props.location.pathname.split('/')[2],
+        oldExpertValue: this.props.location.pathname.split('/')[2],
+        commitId: this.props.location.pathname.split('/')[5],
+        groupSelect: false,
+        languageSelect: false,
+        expertSelect: false,
+        date: '',
+        author: '',
+        oldImageUrl: '',
+        imageUrl: '<image_url>',
+        image_name_changed: false,
+        fontSizeCode: 14,
+        editorTheme: 'github',
+        showAdmin: false,
+        deleteDisabled: true,
+        slideState: 1, // 1 means in middle, 2 means preview collapsed
+        colSkill: this.props.hasOwnProperty('revertingCommit') ? 12 : 8,
+        colPreview: this.props.hasOwnProperty('revertingCommit') ? 0 : 4,
+        prevButton: 0, // 0 means disappear, 1 means appear
+      };
+      let fonts = [14, 16, 18, 20, 24, 28, 32, 40];
```

Comment:
Make a constants file under SkillCreator folder and declare the constants there and then import and use it here